server/command/command.go

```diff
@@ -0,0 +1,63 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/pluginapi"
+)
+
+type Handler struct {
+	client *pluginapi.Client
+}
+
+type Command interface {
+	Handle(args *model.CommandArgs) (*model.CommandResponse, *model.AppError)
+	executeHelloCommand(args *model.CommandArgs) *model.CommandResponse
+}
+
+const helloCommandTrigger = "hello"
+
+// Register all your slash commands in the NewCommandHandler function.
+func NewCommandHandler(client *pluginapi.Client) Command {
```

Comment:
One day, it would be awesome to hoist this into the `public` package and just register callbacks for the slash commands to be handled.

In that future, I wonder if we would need a discrete package at all and could just do this all in `command.go`?